cpp/src/io/parquet/page_decode.cuh

```diff
@@ -506,14 +506,9 @@ __device__ void gpuDecodeStream(
             cur++;
           }
         }
-        if (cur > end) {
-          s->set_error_code(decode_error::LEVEL_STREAM_OVERRUN);
-          break;
-        }
-        if (level_run <= 1) {
-          s->set_error_code(decode_error::INVALID_LEVEL_RUN);
-          break;
-        }
+        // If there are errors, set the error code and continue. The loop will be exited below.
+        if (cur > end) { s->set_error_code(decode_error::LEVEL_STREAM_OVERRUN); }
+        if (level_run <= 1) { s->set_error_code(decode_error::INVALID_LEVEL_RUN); }
```

Comment:
Does this need to be an else if?  In the original code, if cur > end then the error would be LEVEL_STREAM_OVERRUN and we would never check level_run.  In this code, we will fall through to the level_run check, and it looks like level_run won't be updated if cur > end, leaving level_run <= 1.  That seems to indicate that a LEVEL_STREAM_OVERRUN error will likely be smashed by INVALID_LEVEL_RUN unless cur == end.

Reply:
The error codes are flag bits, so if both `cur > end` and `level_run <= 1` are true, then both bits will be set in the final `error_code` mask.

Reply:
Looks like the name `set_error_code` is a bit misleading here :(

Reply:
Sorry, missed that these were bits and not values that would clobber.   But it still looks like this will change the error bits set from the previous code when LEVEL_STREAM_OVERRUN occurs, and I'm wondering if that's intentional.

Reply:
The original original code had
```
if (cur > end || level_run <= 1) { s->error = 0x10; }
```
So both errors shared a single code. #14167 split that up into separate errors, but thread 0 exited too soon (leading to the odd compiler behavior). Here we can capture both errors if both occur, so yes, it's intentional not to do the `else`.